course/models.py

```diff
@@ -564,9 +575,13 @@ def __str__(self):
 
     def trigger(self, data):
         logger = logging.getLogger("plus.hooks")
+        url, data = url_with_query_in_data(self.hook_url, data)
         try:
-            urllib.request.urlopen(self.hook_url,
-                urllib.parse.urlencode(data).encode('utf-8'), timeout=10)
+            urllib.request.urlopen(
+                url,
+                urllib.parse.urlencode(data).encode('ascii'),
+                timeout=10,
+            )
```

Comment:
Interesting, I just noticed that we use urllib here and not requests, like we do in every other place. It's fine, just an observation.